### PR TITLE
rbd: implement deleteImage() with go-ceph

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

Use go-ceph librdb bindings to delete an RBD image in case CephMgr does not support background deletion.

### CODE REVIEW

**Comment by humblec** [comment]
@Madhu-1 @ShyamsundarR ptal.

**Comment by nixpanic** [comment]
Marking Do-Not-Merge for now. Reworking the connection stuff so that it can be shared between rbd/cephfs/rados requests.

**Comment by nixpanic** [comment]
This should be ready for review now, @Madhu-1

File: internal/rbd/rbd_util.go [hunk 1]
```diff
@@ -288,12 +287,17 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
...
-			"--keyfile=" + cr.KeyFile}
-		output, err = execCommand("rbd", args)
+		var ioctx *rados.IOContext
+		ioctx, err = pOpts.conn.GetIoctx(pOpts.Pool)
+		if err != nil {
+			return err
+		}
+		defer ioctx.Destroy()
+
+		image := librbd.GetImage(ioctx, pOpts.RbdImageName)
```
Comment (Madhu-1) [inline internal/rbd/rbd_util.go:297]: we already have a variable called image here https://github.com/ceph/ceph-csi/blob/1f0f2c04e12bb931b8d634aace760d73d2a75a0d/internal/rbd/rbd_util.go#L270. rename this to something else

Comment (nixpanic) [reply internal/rbd/rbd_util.go:297]: Interesting, the linters are so strict, and mostly complain about these kind of things. Somehow this sneaked through!

**Pull request merged**